cpp/models/abm/person.cpp

```diff
@@ -137,6 +140,20 @@ bool Person::goes_to_work(TimePoint t, const AbmMigrationParameters& params) con
     return m_random_workgroup < params.get<WorkRatio>().get_matrix_at(t.days())[0];
 }
 
+int Person::get_go_to_work_hour(const AbmMigrationParameters& params) const
+{
+    int Minimum_Goto_Work_time = params.get<GotoWorkTimeMinimum>()[m_age];
+    int Maximum_Goto_Work_time = params.get<GotoWorkTimeMaximum>()[m_age];
+    return Minimum_Goto_Work_time+(int) ((Maximum_Goto_Work_time-Minimum_Goto_Work_time+1)*m_random_goto_work_hour);
```

Comment:
If I understand this correctly, you are adding `1` to `max - min` so the result after integer truncating can be `max`. I think there is a better way to do this. Also, the time step of the simulation may not be fixed anyway, so rounding to whole hours may cause problems.

I would suggest doing the calculation using `TimeSpan`, which has a better resolution than hours and is semantically clearer. so `t_work = tmin + (tmax - tmin) * r`. Then either go to work if `t + 0.5 * dt >= t_work` or if `t >= t_work`, not sure which is better, just a question of which rounding to use. probably `t + 0.5 * dt >= t_work` is better, since the expected value will be `(max + min) * 0.5`.